My first commit

Mikhail-Besedin opened the pull request:
Я не спал нормально 2 дня, т.к. мне снились решения как написать данный проект. 
Очень жаль, что самой теории не так много . И что то вообще пришлось гуглить. 

Надеюсь не все так плохо, как мне кажется .

Mikhail-Besedin committed:
My first commit

AlexKitIt commented:
Гуглить теперь - это на всю жизнь если есть планы работать и развиваться в профессии) Всю теорию на блюде не принесет ни один курс. Одна из задач (и далеко не самая низкоприоритетная) это умение разбираться в ситуации имея минимум вводной информации или не имея ее вовсе

AlexKitIt commented on pages/main_page.py:
```diff
@@ -0,0 +1,18 @@
+
+    @allure.step(' Скролим до элемента ')
+    def scroll_to_element(self, locator):
+        element = self.driver.find_element(*locator)
+        self.driver.execute_script("arguments[0].scrollIntoView();", element)
```
Нужно исправить: здесь и далее по проекту: все прямые обращения к driver и waitы необходимо вынести в base и оттуда за счет наследования переиспользовать в объектах страниц

AlexKitIt commented on pages/order_page.py:
```diff
@@ -0,0 +1,44 @@
+class OrderPage(BasePage):
+    @allure.step('''Заполняем данные на странице "для кого самокат" и "Про аренду" и кликаем на кнопку "Заказать",
+     получая текст страницы успешного заказа ''')
+    def set_order(self,name_locator, name,
```
Можно лучше: метод стоит разделить на два, там две отдельные формы и будет проще определять что именно сломалось

AlexKitIt commented on tests/test_main_page.py:
```diff
@@ -0,0 +1,76 @@
+    )
+    def test_questions_and_answer(self, driver, num, result):
+        main_page = MainPage(driver)
+        main_page.scroll_to_element(MainPageLocators.BLOCK_OF_QUESTION_LOCATOR)
```
Нужно исправить: здесь и далее: локаторов в тестах быть не должно, если это не вариант параметризации, они живут в методах пейджи, тест должен быть максимально изолирован от логики выполнения

AlexKitIt commented on tests/test_main_page.py:
```diff
@@ -0,0 +1,76 @@
+    def test_ya_logo(self,driver):
+        main_page = MainPage(driver)
+        main_page.click_to_element(MainPageLocators.YA_LOGO)
+        driver.switch_to.window(driver.window_handles[1])
```
Нужно исправить: прямого обращения к driver из тестов быть не должно , только через методы basePage

AlexKitIt commented on tests/test_main_page.py:
```diff
@@ -0,0 +1,76 @@
+
+
+
+    @allure.title('Тест проверки перехода на главную страницу веб-приложения по клику на логотип "Самокат"')
+    @allure.description('''1)кликаем на кнопку "Заказать"
+                                2)Кликаем на логотип "Самокат"
+                                3)Проверяем наличие текста в описании заголовка  с ожидаемым  заголовком ''')
+    def test_logo_scooter(self, driver):
+        main_page = MainPage(driver)
+        main_page.click_to_element(MainPageLocators.HEADER_ORDER_BUTTON)
+        main_page.click_to_element(MainPageLocators.LOGO_SCOOTER)
+        assert 'пару дней'in main_page.get_text_from_element(MainPageLocators.TITLE_HOME_LOCATOR)
+
+    @allure.title('Тест проверки перехода на Яндекс.Дзен веб-приложении по клику на логотип "Яндекса"')
+    @allure.description('''1)кликаем на кнопку "Яндекс"
+                        2)Переходим на новую страницу браузера в Яндекс.Дзен
+                        3)Проверяем отображение элемента "Главная" для подтверждения перехода ''')
+    def test_ya_logo(self,driver):
```
Нужно исправить: тесты редиректов необходимо вынести в отдельный модуль и класс. Они объединяются по общей функциональности а не странице, на main может быть и 1000 тестов, не в одном же модуле их хранить

AlexKitIt commented on tests/test_order_page.py:
```diff
@@ -0,0 +1,45 @@
+    def test_place_an_order(self, driver,name, last_name, adress,tel,comment):
+        order_page = OrderPage(driver)
+        order_page.click_to_element(MainPageLocators.HEADER_ORDER_BUTTON)
+        assert 'Заказ оформлен' in order_page.set_order(OrderPageLocators.NAME_LOCATOR, name,
```
Можно лучше: саму проверку стоит описать отдельно, а то выглядит очень громоздко

Mikhail-Besedin committed:
Будет перенесен метод scroll_to_element() в base_page из main_page

Mikhail-Besedin committed:
Будет разделен метод set_order() в order_page на два разных метода set_order_to_the_next_button() и set_order_to_the_order_button(). Так будет проще определять что именно сломалось.

Mikhail-Besedin committed:
Будут убраны все локаторы из всех файлов с тестами.
Локаторы перемещены в файлы страниц.
В связи с тем что локаторы перемещались в методы страниц, пересобирались сами методы и тесты.

В файл base_page.py - добавлены методы :
switch_to_window() - переключение на другое окно браузера

Появятся файлы :
redirect_page - страница с методами редиректа;
redirect_page_locators -  локаторы страницы редиректа;
test_redirect_page.py - тесты для страницы редиректа.

Mikhail-Besedin committed:
Будет создан новая директория с отчетами Allure.

Mikhail-Besedin commented:
Я переписал почти весь проект заново . 
Надеюсь теперь все хорошо, и я могу идти дальше учиться )))
